src/task_monitor/tasks/insert_identities.rs

```diff
@@ -42,62 +40,36 @@ pub async fn insert_identities(
         let _guard = pending_insertions_mutex.lock().await;
         let latest_tree = app.tree_state()?.latest_tree();
 
-        retry_tx!(&app.database, tx, {
-            insert_identities_batch(&mut tx, latest_tree, &unprocessed).await
-        })
-        .await?;
+        let mut tx = app.database.begin_tx(IsolationLevel::ReadCommitted).await?;
 
-        // Notify the identity processing task, that there are new identities
-        wake_up_notify.notify_one();
-    }
-}
-
-pub async fn insert_identities_batch(
-    tx: &mut Transaction<'_, Postgres>,
-    latest_tree: &TreeVersion<Latest>,
-    identities: &[UnprocessedCommitment],
-) -> Result<(), anyhow::Error> {
-    // Filter out any identities that are already in the `identities` table
-    let mut filtered_identities = vec![];
-    for identity in identities {
-        if tx
-            .get_identity_leaf_index(&identity.commitment)
-            .await?
-            .is_some()
-        {
-            tracing::warn!(?identity.commitment, "Duplicate identity");
-            tx.remove_unprocessed_identity(&identity.commitment).await?;
-        } else {
-            filtered_identities.push(identity.commitment);
-        }
-    }
+        let next_leaf = latest_tree.next_leaf();
+        let mut pre_root = latest_tree.get_root();
 
-    let next_leaf = latest_tree.next_leaf();
+        let next_db_index = tx.get_next_leaf_index().await?;
+        assert_eq!(
+            next_leaf, next_db_index,
+            "Database and tree are out of sync. Next leaf index in tree is: {next_leaf}, in database: {next_db_index}"
+        );
 
-    let next_db_index = tx.get_next_leaf_index().await?;
+        for (idx, identity) in unprocessed.iter().enumerate() {
+            let leaf_idx = next_leaf + idx;
+            latest_tree.update(leaf_idx, *identity);
+            let root = latest_tree.get_root();
 
-    assert_eq!(
-        next_leaf, next_db_index,
-        "Database and tree are out of sync. Next leaf index in tree is: {next_leaf}, in database: \
-         {next_db_index}"
-    );
+            tx.insert_pending_identity(leaf_idx, identity, &root, &pre_root)
+                .await
+                .expect("Failed to insert identity - tree will be out of sync");
```

Comment:
The most critical change in this PR is this and the other expect on line 70.

These are necessary to keep the sequencer state in sync with the db

Reply:
I guess my only question is will the `identity` be completely lost here if this fails? And therein all unprocessed identities prior? Same below with `tx.commit()` failing. Just wondering what the flow would be after restart